packages/dataviews/src/view-list.tsx

```diff
@@ -142,23 +205,137 @@ function ListItem( {
 						</HStack>
 					</CompositeItem>
 				</div>
+				{ actions?.length > 0 && (
```

Comment:
I don't understand why we can't reuse item-actions here? (Maybe with a new prop that indicates that we should only use one primary action). What's the other differences between this and item actions?

Reply:
These are the differences: every button trigger needs to be wrapped by a `CompositeItem` the receives the composite store, the "all actions" button needs to handle `onkeydown` event on their own, it only render 1 primary action.

An alternative to make `ItemActions` work for this layout would be adding `ifs` everywhere, though I discarded it because it is too intertwined. When `table` and `grid` layout improve the keyboard interactions, `ItemActions` would actually be more reusable as all the layouts would share the same logic.

Reply:
> When table and grid layout improve the keyboard interactions, ItemActions would actually be more reusable as all the layouts would share the same logic.

With the gained ariakit/composite experience, I actually want to tackle that next, though it'll probably land for 6.7.

Reply:
> An alternative to make ItemActions work for this layout would be adding ifs everywhere, though I discarded it because it is too intertwined. When table and grid layout improve the keyboard interactions, ItemActions would actually be more reusable as all the layouts would share the same logic.

How can we make sure we actually do that and not end up with two similar components.

Reply:
> How can we make sure we actually do that and not end up with two similar components.

We'll have more info about the proper abstractions for the actions component when keyboard interactions are improved for other layouts. The existing proved not enough to use with Composite, for example. Personally, I won't be able to start looking into this until end of May/first of June, the soonest. If we'd rather have a single component that accommodates all the possible use cases in the interim, I'm happy to switch the implementation.